src/karapace/api/oidc/middleware.py

```diff
@@ -43,6 +49,29 @@ def __init__(self, app: FastAPI, config: Config) -> None:
             )
 
             self._jwks_client = PyJWKClient(self.jwks_url)
+
+            log.info("OIDC Authorization enabled: %s", self.authorization_enabled)
+            if self.authorization_enabled:
+                if self.client_id is None or self.sasl_oauthbearer_roles_claim_path is None:
+                    raise HTTPException(
+                        status_code=403,
+                        detail="Invalid configuration: client_id and roles_claim_path are required when authorization is enabled.",
+                    )
+
+                required_http_methods = {"GET", "POST", "PUT", "DELETE"}
```

Comment:
Maybe this should come from the keys of `self.sasl_oauthbearer_method_roles`, that way we are not forcing them to use auth on methods they haven't configured, unless we want to enforce auth on all these methods if OIDC is enabled.

Reply:
Done. Good one.